### PR TITLE
build: refactor external packages

### PR DESCRIPTION
**Pull request by bassam**

### CODE REVIEW

File: build/makelib/cephd.mk [hunk 1]
```diff
@@ -43,6 +43,7 @@ CEPHD_CMAKE += \
 	-DALLOCATOR=$(CEPHD_ALLOCATOR) \
 	-DWITH_FUSE=OFF \
 	-DWITH_NSS=OFF \
+	-DWITH_SSL=OFF \
```
Comment (jbw976) [inline build/makelib/cephd.mk]: why did these flags change? (curious about the KRBD one too)

File: Makefile [hunk 2]
```diff
@@ -178,37 +143,62 @@ include build/makelib/golang.mk
 
 RELEASE_VERSION=$(VERSION)
 RELEASE_BIN_DIR=$(BIN_DIR)
-RELEASE_CLIENT_SERVER_PLATFORMS=$(CLIENT_SERVER_PLATFORMS)
-RELEASE_CLIENT_ONLY_PLATFORMS=$(CLIENT_ONLY_PLATFORMS)
```
Comment (jbw976) [inline Makefile]: do we no longer have a concept of client only platforms? i didn't see that this was moved to somewhere else from a quick search.

File: build/makelib/golang.mk [hunk 3]
```diff
@@ -128,30 +128,36 @@ go.init: $(GO_VENDOR_DIR)/vendor.stamp
...
-.PHONY: go.build
-go.build: go.vet go.fmt $(CGO_PREREQS)
-	@$(foreach p,$(GO_NONSTATIC_PACKAGES),CGO_ENABLED=1 $(GO) build -v -i -o $(GO_OUT_DIR)/$(lastword $(subst /, ,$(p)))$(GO_OUT_EXT) $(GO_NONSTATIC_FLAGS) $(p);)
-	@$(foreach p,$(GO_NONSTATIC_PIE_PACKAGES),CGO_ENABLED=1 $(GO) build -v -i -o $(GO_OUT_DIR)/$(lastword $(subst /, ,$(p)))$(GO_OUT_EXT) $(GO_NONSTATIC_PIE_FLAGS) $(p);)
-	@$(foreach p,$(GO_STATIC_PACKAGES),CGO_ENABLED=0 $(GO) build -v -i -o $(GO_OUT_DIR)/$(lastword $(subst /, ,$(p)))$(GO_OUT_EXT) $(GO_STATIC_FLAGS) $(p);)
-	@$(foreach p,$(GO_STATIC_CGO_PACKAGES),CGO_ENABLED=1 $(GO) build -v -i -o $(GO_OUT_DIR)/$(lastword $(subst /, ,$(p)))$(GO_OUT_EXT) $(GO_STATIC_CGO_FLAGS) $(p);)
-ifeq ($(RELEASEBUILD),1)
+define go.project
+go.build.packages.$(1):
+	@CGO_ENABLED=$(3) $(GO) build -v -i -o $(GO_OUT_DIR)/$(1)$(GO_OUT_EXT) $(4) $(2)
```
Comment (jbw976) [inline build/makelib/golang.mk]: i haven't figured out what all these numbered params are yet, where would i look for that?

File: external/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ccache can significantly speed up the build. Its turned on by default. To disable:
+
+```
+make CCACHE=0
+```
+
+You can also build the packages inside our cross build container:
+
+```
+build/run make -C external
```
Comment (jbw976) [inline external/README.md]: Since this readme is in the external dir, would this be `../build/run` instead?

File: build/makelib/golang.mk (hunk 3 above)
Comment (bassam) [inline build/makelib/golang.mk]: they are the params passed through `call` invocation

File: Makefile (hunk 2 above)
Comment (bassam) [inline Makefile]: we do however, the client vs server determination now happens in the release scripts

File: external/README.md (hunk 4 above)
Comment (bassam) [inline external/README.md]: I was assuming this was in the root of the tree

**Comment by jbw976** [comment]
Is there any migration with the new submodules that a developer would need to worry about? or is it as simple as:
```
git submodule sync --recursive
git submodule update --init --recursive
```

**Comment by bassam** [comment]
Yes certainly update the submodules,  and I recommend building clean `make distclean` and maybe even `build/reset`.

**Pull request merged**